source/agora/flash/Network.d

```diff
@@ -0,0 +1,331 @@
+    {
+        const chan_id = chan_conf.chan_id;
+
+        if (peer1_pk !in this.nodes)
+            this.nodes[peer1_pk] = NetworkNode.init;
+
+        if (auto chns = (peer2_pk in this.nodes[peer1_pk].channels))
+            chns.put(chan_id);
+        else
+            this.nodes[peer1_pk].channels[peer2_pk] = Set!Hash.from([chan_id]);
+    }
+
+    /***************************************************************************
+
+        Add multiple Channels to the network
+
+        Params:
+            chns = List of ChannelConfig
+
+    ***************************************************************************/
+
+    public void addChannels (const ref ChannelConfig[] chns) @safe nothrow
+    {
+        foreach (chn; chns)
+            this.addChannel(chn);
+    }
+
+    /***************************************************************************
+
+        Remove a Channel from the network
+
+        Params:
+            chan_conf = ChannelConfig of the Channel to be removed
+
+    ***************************************************************************/
+
+    public void removeChannel (const ref ChannelConfig chan_conf) @safe nothrow
+    {
+        const funder_pk = chan_conf.funder_pk;
+        const peer_pk = chan_conf.peer_pk;
+
+        this.removeChannel(funder_pk, peer_pk, chan_conf);
+        this.removeChannel(peer_pk, funder_pk, chan_conf);
+    }
+
+    private void removeChannel (Point peer1_pk, Point peer2_pk,
+        const ref ChannelConfig chan_conf) @safe nothrow
+    {
+        const chan_id = chan_conf.chan_id;
+
+        // Remove channels
+        this.nodes[peer1_pk].channels[peer2_pk].remove(chan_id);
+
+        // If no channels remain, remove the peer
+        if (this.nodes[peer1_pk].channels[peer2_pk].length == 0)
+            this.nodes[peer1_pk].channels.remove(peer2_pk);
+
+        // If no peers remain, remove the node
+        if (this.nodes[peer1_pk].channels.length == 0)
+            this.nodes.remove(peer1_pk);
+    }
+
+    /***************************************************************************
+
+        Remove multiple Channels from the network
+
+        Params:
+            chns = List of ChannelConfig
+
+    ***************************************************************************/
+
+    public void removeChannel (const ref ChannelConfig[] chns) @safe nothrow
+    {
+        foreach (chn; chns)
+            this.removeChannel(chn);
+    }
+
+    /***************************************************************************
+
+        Build a path between two nodes in the network
+
+        Params:
+            from_pk = Source node public key
+            to_pk = Destination node public key
+            ignore_chans = Channels to ignore
+
+        Returns:
+            If found, path from source to destination
+
+    ***************************************************************************/
+
+    public Hop[] getPath (Point from_pk, Point to_pk,
```

Comment:
I would name this `getPaymentPath`. We will add the `Amount` later, so it will make more sense.

Or you could choose to rename it later when the Amount is added? I think that's fine too.